soepy/python/pre_processing/model_processing.py

```diff
@@ -13,154 +18,233 @@ def read_init_file(init_file_name):
     else:
         init_dict = init_file_name
 
-    init_dict = expand_init_dict(init_dict)
+    # Determine categories
+    category = []
 
-    model_params = create_namedtuple(init_dict)
+    for (key, value) in init_dict["PARAMETERS"].items():
+        # Check if key is even then add pair to new dictionary
+        if "gamma_0" in key:
+            category.append("const_wage_eq")
+        elif "gamma_1" in key:
+            category.append("exp_returns")
+        elif "g_s" in key:
+            category.append("exp_accm")
+        elif "delta" in key:
+            category.append("exp_deprec")
+        elif "const" in key:
+            category.append("disutil_work")
+        elif "theta" in key:
+            category.append("hetrg_unobs")
+        elif "share" in key:
+            category.append("shares")
+        elif "sigma" in key:
+            category.append("sd_wage_shock")
 
-    return model_params
+    # Create data frame
+    columns = ["name", "value"]
 
+    data = list(init_dict["PARAMETERS"].items())
 
-def expand_init_dict(init_dict):
-    """Enhances read in initialization dictionary by
-    adding model parameters derived from the
-    specified initialisation file"""
+    model_params_df = pd.DataFrame(data, columns=columns)
 
-    # Calculate range of years of education in the (simulated) sample
-    educ_min = init_dict["INITIAL_CONDITIONS"]["educ_min"]
-    educ_max = init_dict["INITIAL_CONDITIONS"]["educ_max"]
-    educ_range = educ_max - educ_min + 1
+    model_params_df.insert(0, "category", category, True)
 
-    # Calculate covariances of the error terms given standard deviations
-    shocks_cov = [
-        init_dict["PARAMETERS"]["sigma_1"],
-        init_dict["PARAMETERS"]["sigma_2"],
-        init_dict["PARAMETERS"]["sigma_3"],
-    ]
-    shocks_cov = [shocks_cov[0] ** 2, shocks_cov[1] ** 2, shocks_cov[2] ** 2]
+    model_params_df.set_index(["category", "name"], inplace=True)
 
-    # Extract the number of types
-    type_shares_non_baseline = [
-        _ for k, _ in init_dict["PARAMETERS"].items() if "share" in k
-    ]
-    num_types = len(type_shares_non_baseline) + 1
+    return model_params_df
 
-    # Aggregate type shares in list object
-    # Share of baseline types equal to one minus sum of remaining type shares
-    type_shares = [1 - sum(type_shares_non_baseline)] + type_shares_non_baseline
 
-    # Append derived attributes to init_dict
-    init_dict["DERIVED_ATTR"] = {
-        "educ_range": educ_range,
-        "shocks_cov": shocks_cov,
-        "num_types": num_types,
-        "type_shares": type_shares,
-    }
+# Pre-processing of model parameters to be estimated
```

Comment:
Please remove or explain why it should be there.

Reply:
Are you referring to the line with the comment?